data/README.md

```diff
@@ -1,132 +1,199 @@
 
-```js
-wp.data.dispatch( 'myPlugin' ).setTitle( 'new Title' ); // Dispatches the setTitle action to the reducer
-```
+#### `withSelect( mapSelectToProps: Function ): Function`
 
-### `wp.data.subscribe( listener: function )`
+Use `withSelect` to inject state-derived props into a component. Passed a function which returns an object mapping prop names to the subscribed data source, a higher-order component function is returned. The higher-order component can be used to enhance a presentational component, updating it automatically when state changes. The mapping function is passed the [`select` function](#select) and the props passed to the original component.
 
-Function used to subscribe to data changes. The listener function is called each time a change is made to any of the registered reducers. This function returns a `unsubscribe` function used to abort the subscription.
+__Example:__
 
 ```js
-// Subscribe.
-const unsubscribe = wp.data.subscribe( () => {
-	const data = {
-		slug: wp.data.select( 'core/editor' ).getEditedPostSlug(),
-	};
+function PriceDisplay( { price, currency } ) {
+	return new Intl.NumberFormat( 'en-US', {
+		style: 'currency',
+		currency,
+	} ).format( price );
+}
 
-	console.log( 'data changed', data );
-} );
+const { withSelect } = wp.data;
 
-// Unsubcribe.
-unsubscribe();
+const HammerPriceDisplay = withSelect( ( select, ownProps ) => {
+	const { getPrice } = select( 'my-shop' );
+	const { currency } = ownProps;
+
+	return {
+		price: getPrice( 'hammer', currency ),
+	};
+} )( PriceDisplay );
+
+// Rendered in the application:
+//
+//  <HammerPriceDisplay currency="USD" />
 ```
 
-### `wp.data.withSelect( mapStateToProps: Object|Function )( WrappedComponent: Component )`
+In the above example, when `HammerPriceDisplay` is rendered into an application, it will pass the price into the underlying `PriceDisplay` component and update automatically if the price of a hammer ever changes in the store.
 
-To inject state-derived props into a WordPress Element Component, use the `withSelect` higher-order component:
+#### `withDispatch( mapDispatchToProps: Function ): Function`
+
+Use `withDispatch` to inject dispatching action props into your component. Passed a function which returns an object mapping prop names to action dispatchers, a higher-order component function is returned. The higher-order component can be used to enhance a presentational component, often providing callback behaviors for user interactions. The mapping function is passed the [`dispatch` function](#dispatch) and the props passed to the original component.
```

Comment:
Technically `withSelect` and `withDispatch` can enhance any component and not only presentational components.

Reply:
>Technically `withSelect` and `withDispatch` can enhance any component and not only presentational components.

The thinking was to build upon this idea that visual components become enhanced to react to state with these higher-order components. Not to imply a sort of "rule" about when they can be used.

Do you have an example in mind that contradicts this advice?